TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/ScoringAlgorithm.java

```diff
@@ -107,25 +110,25 @@ public void resetPos() {
         telemetry.addData("Scoring Status1` ", "Resetting Pos");
 
         robot.<Servo>get("jointServo").setPosition(jointStartPos);
-        robot.<DcMotor>get("armMotor").setDirection(DcMotor.Direction.REVERSE);
-        robot.<DcMotor>get("leftViperSlideMotor").setDirection(DcMotorSimple.Direction.REVERSE);
-        robot.<DcMotor>get("rightViperSlideMotor").setDirection(DcMotorSimple.Direction.REVERSE);
-
-        while (robot.<DcMotor>get("armMotor").getCurrentPosition() != armStartPos) {
-
-            if (robot.<DcMotor>get("rightViperSlideMotor").getCurrentPosition() != viperSlideStartPos) {
-                robot.<DcMotor>get("rightViperSlideMotor").setPower(0.6);
-            } else if (robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() == viperSlideStartPos) {
-                robot.<DcMotor>get("leftViperSlideMotor").setPower(0);
-            } else if (robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() != viperSlideStartPos) {
-                robot.<DcMotor>get("leftViperSlideMotor").setPower(0.6);
-            } else if (robot.<DcMotor>get("leftViperSlideMotor").getCurrentPosition() == viperSlideStartPos) {
-                robot.<DcMotor>get("leftViperSlideMotor").setPower(0);
-            }
 
+        while (Math.abs(robot.<DcMotor>get("armMotor").getCurrentPosition() - armStartPos) > armStartTolerance) {
```

Comment:
same thing as above

> there's a chance you might want to add a condition to catch if it would ever miss your entire section. you can solve this by either changing the while condition to be while (robot.<DcMotor>get("armMotor").getCurrentPosition < armTargetPos){} which would work well to catch it as soon as it passes that threshold
> 
> however, if you're more concerned with accuracy and willing to sacrifice some time, you can do something that's basically inside of your while loop that checks if (robot.<DcMotor>get("armMotor").getCurrentPosition < armTargetPos - armTolerance){} and add an else if to check else if (robot.<DcMotor>get("armMotor").getCurrentPosition > armTargetPos + armTolerance){} which would basically let it self correct

Reply:
see above....